eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/NotificationRendererPass.php

```diff
@@ -16,20 +17,47 @@
  */
 class NotificationRendererPass implements CompilerPassInterface
 {
-    /**
-     * @param \Symfony\Component\DependencyInjection\ContainerBuilder $container
-     */
+    const TAG_NAME = 'ezpublish.notification.renderer';
+    const REGISTRY_DEFINITION_ID = 'notification.renderer.registry';
+
     public function process(ContainerBuilder $container)
     {
-        if (!$container->has('notification.renderer.registry')) {
+        if (!$container->has(self::REGISTRY_DEFINITION_ID)) {
             return;
         }
 
-        $registry = $container->findDefinition('notification.renderer.registry');
+        $registry = $container->findDefinition(self::REGISTRY_DEFINITION_ID);
+
+        foreach ($container->findTaggedServiceIds(self::TAG_NAME) as $id => $attributes) {
+            foreach ($attributes as $attribute) {
+                if (!isset($attribute['alias'])) {
+                    throw new LogicException(sprintf(
+                        'Tag %s needs a "alias" attribute to identify the notification type. None given.',
+                        self::TAG_NAME
+                    ));
+                }
+
+                $registry->addMethodCall('addRenderer', [$attribute['alias'], new Reference($id)]);
+            }
+        }
+
+        foreach ($container->findTaggedServiceIds('ezstudio.notification.renderer') as $id => $attributes) {
```

Comment:
I think with current list of BC breaks we don't have to care about that one ;)

Reply:
> I think with current list of BC breaks we don't have to care about that one ;)

... Yes we DO care about that. If EE Customer used Cron Bundle before, he'll get hit hard by that, even more than by totally changed namespaces, assuming he's not doing something crazy.

Besides, the fact that we did BC breaks when moving this doesn't mean that we're gonna do it now everywhere. If we can - we ensure BC. Really. Please change your mindset - this is what actually worries me, not some probably not widely used renderer.

Reply:
Ok, understood. btw, how is that related to CronBundle?

Reply:
Meant Notifications, not Cron of course, too many things at once, sorry ;)

Reply:
> I think with current list of BC breaks we don't have to care about that one ;)

INHO keeping BC compatibility in case of `ezstudio.notification.renderer` tag doesn't make sense when we didn't keep it in other layers.

Reply:
I agree. Nobody is using it. Please proceed with this.

Reply:
> INHO keeping BC compatibility in case of ezstudio.notification.renderer tag doesn't make sense when we didn't keep it in other layers.

My point was about general approach, decided to comment in this case because you can tag your service and even not being aware that something has changed. That is - if your tagged class doesn't internally rely on any of other BC breaking changes. Don't know if that does make sense here indeed, but still... :)